### ISSUE TITLE
Need a Haikuports recipe style guide added to the wiki

### ISSUE THREAD

**Issue by scottmc** [haikuports/haikuports#3223]
I keep seeing the same types of comments on pull requests.  Instead of pointing them out each time, we should be able to just point recipe authors to the style guide, but we don't have one (yet?).

**Comment by scottmc** [comment]
Page started:
https://github.com/haikuports/haikuports/wiki/HaikuPorter-StyleGuide

**Comment by humdingerb** [comment]
The "Guidelines" at https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines contains already much style related info. It may be useful to separate style and general guidelines (and add a link to the style guide from the general guidelines).

**Comment by humdingerb** [comment]
On second thought... often a style-related item has a more general aspect that's explained as a side-note. It makes sense to keep all in one document after all.

**Comment by scottmc** [comment]
You are correct.  Not sure why I didn't see that page before, seems to cover most things.  If anyone spots any missing Style related issues, please bring it up or update that wiki page.  I have deleted the proposed page, as the guidelines page works.

**Comment by Begasus** [comment]
I linked the page a few days back in the GCI mailing list Scot :)

Op wo 17 okt. 2018 om 07:42 schreef Scott McCreary <notifications@github.com

**Comment by Begasus** [comment]
On the other hand, I don't think there is a page or guideline for the
makefiles that uses the makefile-engine? Maybe something could be done
there (as most of the HaikuArchives will rely on those)

Op wo 17 okt. 2018 om 08:50 schreef Luc Schrijvers <begasus@gmail.com>:

**Comment by humdingerb** [comment]
> On the other hand, I don't think there is a page or guideline for the makefiles that uses the makefile-engine? Maybe something could be done there (as most of the HaikuArchives will rely on those)

Aren't our makefile-engine's makefiles already quite elaborately annotated?

**Comment by Begasus** [comment]
Not all projects on HaikuArchives already have a makefile, some have a
.proj file and with some luck you can generate one from it (not documented
too), some don't even have a .proj file nor makefie, so they should be
created from scratch. The default makefile (clean one) isn't documented
either (or maybe I missed something and it's documented somewhere I haven't
been yet), a first time user probably wouldn't know where to look for it. :)

Op wo 17 okt. 2018 om 09:29 schreef humdinger <notifications@github.com>:

**Comment by humdingerb** [comment]
> The default makefile (clean one) isn't documented either (or maybe I missed something and it's documented somewhere I haven't been yet), a first time user probably wouldn't know where to look for it.

If you create a Makefile from Tracker's "New..." menu, you get this makefile:
https://git.haiku-os.org/haiku/tree/data/develop/Makefile

Not sure where this or how to convert a .proj file (I suppose with Paladin, right?) should be documented.

**Comment by Begasus** [comment]
You just handed 2 nice howto's in your last reply that is not documented
(at least not easy to find if there is one) :)
So it would be just a small wiki page explaining them how to
find/create/edit them, just my 2 cents :)


Op wo 17 okt. 2018 om 10:04 schreef humdinger <notifications@github.com>:

**Comment by humdingerb** [comment]
The makefile-engine's Makefile is a bit meager for a whole wiki page. Maybe if some of the porting gurus can give hints to other buildsystems, like cmake etc.

**Comment by alaviss** [comment]
@fbrosson I think we should clarify the styling of `LICENSE` vs `BUILD_REQUIRES` variables in the Guidelines.
